Update library overview [8399] (#118)
eProsima/Fast-RTPS-docs: docs/fastdds/library_overview/includes/architecture.rst

* **DomainId_t**.
  uint32_t data type which identifies the DDS domain.
  Each DomainParticipant will have an assigned DDS domain, so that DomainParticipants in the same domain can
  communicate, as well as isolate communications between DDS domains.
  This value must be given by the application developer when creating the DomainParticipants.
* **DomainParticipants**.
  Object containing other DDS entities such as publishers, subscribers, topics and multitopics.
  It is the entity that allows the creation of the previous entities it contains, as well as the configuration of their
  behavior.
* **Publisher**.
  It is the entity that creates and configures the DataWriter entities it contains.
  It may contain one or more DataWriter entities.
* **DataWriter**.
  It is the entity in charge of publishing messages.
  The user must provide a Topic when creating this entity which will be the Topic under which the data will be
  published.
  Publication is done by writing the data-objects as a change in the DataWriterHistory.
* **DataWriterHistory**.
  It contains the data objects that are changed by the DataWriter and that are going to be served to the DataReader

>   that subscribes to the topic under which the DataWriter publishes.

Reviewer: I find this a bit confusing. I'd say something like the history is a list of changes in the data in question. Then, I'm thinking that at this point maybe we should have explained somewhere that in DDS the information is considered as updates to a data field

* **Subscriber**.
  It is the entity that creates and configures the DataReader entities it contains.
  It may contain one or more DataReader entities.
* **DataReader**.
  It is the entity that subscribes to the topics for the reception of publications.
  The user must provide a subscription Topic when creating this entity.

>   A DataReader receives the messages as changes in its HistoryDataReader.

Reviewer: Similar to publisher. The subscriber subscribes to a topic using a DataReader, which read the data from the transport

* **DataReaderHistory**.
  It contains the changes in the data-objects made by the DataWriter that publishes under the topic to which the
  DataReader subscribes.
* **Topic**. Entity that binds Publishers' DataWriters with Subscribers' DataReaders.

> The aforementioned **Entities** represent any object that supports QoS, and a listener.

Reviewer: IDK if the definition of entity should come before the list of entities
```suggestion
A DDS **Entity** is any object that supports Quality of Service configuration (QoS), and the implements listener.
```

* **QoS**.
  The mechanism by which the behavior of each of the entities is defined.
* **Listener**.
  The mechanism by which the entities are notified of the possible events that arise during the application's execution.

> This is all covered in more detail in the :ref:`dds_layer` section.

Reviewer: ```suggestion
For a more detailed explanation of each entity, their QoS, and their listeners, please refer to :ref:`dds_layer` section.
```

RTPS layer
^^^^^^^^^^

As mentioned above, the RTPS protocol in Fast DDS allows the abstraction of DDS application entities from the transport
layer.
According to the graph shown above, the RTPS layer has four main **Entities**.

* **RTPSDomain**.
  It is the extension of the DDS domain to the RTPS protocol.
* **RTPSParticipant**.
  Entity containing other RTPS entities. It allows the configuration and creation of the entities it contains.
* **RTPSWriter**.

>   The source of the messages. It reads the changes written in the DataWriterHistory and transmits these changes to the
>   RTPSReader it has previously matched.

Reviewer: ```suggestion
  The source of the messages. It reads the changes written in the DataWriterHistory and transmits them to all
  the RTPSReaders to which it has previously matched.
```

* **RTPSReader**.

>   Receiving entity of the messages. It writes the changes reported by the RTPSWriter in the DataReaderHistory.

Reviewer: ```suggestion
  Receiving entity of the messages. It writes the changes reported by the RTPSWriter into the DataReaderHistory.
```

> This is all covered in more detail in the :ref:`rtps_layer` section.

Reviewer: Same as before

Transport layer
^^^^^^^^^^^^^^^

> Fast DDS supports the implementation of applications over various transport protocols: UDPv4, UDPv6, TCPv4, TCPv6 and

Reviewer: ```suggestion
Fast DDS supports the implementation of applications over various transport protocols. Those are: UDPv4, UDPv6, TCPv4, TCPv6, and
```

A DDS **Entity** is any object that supports Quality of Service configuration (QoS), and the implements listener.

* **QoS**.
  The mechanism by which the behavior of each of the entities is defined.
* **Listener**.
  The mechanism by which the entities are notified of the possible events that arise during the application's execution.

Below are listed the DDS Entities together with their description and functionality.
For a more detailed explanation of each entity, their QoS, and their listeners, please refer to :ref:`dds_layer`
section.

* **Domain**.
  A positive integer which identifies the DDS domain.
  Each DomainParticipant will have an assigned DDS domain, so that DomainParticipants in the same domain can
  communicate, as well as isolate communications between DDS domains.
  This value must be given by the application developer when creating the DomainParticipants.
* **DomainParticipant**.
  Object containing other DDS entities such as publishers, subscribers, topics and multitopics.
  It is the entity that allows the creation of the previous entities it contains, as well as the configuration of their
  behavior.
* **Publisher**.

>   The Publisher publish data under a topic using a DataWriter, which read the data from the transport.

Reviewer: ```suggestion
  The Publisher publishes data under a topic using a DataWriter, which reads the data from the transport.
```

  It is the entity that creates and configures the DataWriter entities it contains, and may contain one or more

>   DataWriter entities.

Reviewer: ```suggestion
  of them.
```

* **DataWriter**.
  It is the entity in charge of publishing messages.
  The user must provide a Topic when creating this entity which will be the Topic under which the data will be
  published.
  Publication is done by writing the data-objects as a change in the DataWriterHistory.
* **DataWriterHistory**.
  This is a list of changes to the data-objects.
  When the DataWriter proceeds to publish data under a specific Topic, it actually creates a `change` in this data.
  It is this `change` that is registered in the History.
  These `changes` are then sent to the DataReader that subscribes to that specific topic.
* **Subscriber**.

>   The Subscriber subscribes to a topic using a DataReader, which read the data from the transport.

Reviewer: ```suggestion
  The Subscriber subscribes to a topic using a DataReader, which reads the data from the transport.
```
